src/content.js

```diff
@@ -37,45 +37,81 @@ async function getTagMetadata(tagsDir) {
   return defsByName;
 }
 
+function getPagePathParts(contentDir, readmeFilePath) {
+  const {dir} = path.parse(readmeFilePath);
+  const contentDirDepth = path.normalize(contentDir).split(path.sep).length;
+  return path.normalize(dir).split(path.sep).slice(contentDirDepth);
+}
+
+//return an array of the metadata objects representing each content page
 async function getPageMetadata(contentDir) {
```

Comment:
These modifications to `getPageMetadata` accomplish a couple things:

* Clearer naming of variables and computed metadata properties for pages. This results in the renames you see in other files
* Add a second pass after the initial file parsing where inter-page relationships are built